tests/bot/exts/moderation/test_silence.py

```diff
@@ -491,3 +748,86 @@ async def test_preserved_other_overwrites(self):
                 del new_overwrite_dict['add_reactions']
 
                 self.assertDictEqual(prev_overwrite_dict, new_overwrite_dict)
+
+    @mock.patch.object(silence.Silence, "_unsilence", return_value=False)
+    @mock.patch.object(silence.Silence, "send_message")
+    async def test_unsilence_helper_fail(self, send_message, _):
+        """Tests unsilence_wrapper when `_unsilence` fails."""
+        ctx = MockContext()
+
+        text_channel = MockTextChannel()
+        text_role = self.cog.bot.get_guild(Guild.id).default_role
+
+        voice_channel = MockVoiceChannel()
+        voice_role = self.cog.bot.get_guild(Guild.id).get_role(Roles.voice_verified)
+
+        test_cases = (
+            (ctx, text_channel, text_role, True, silence.MSG_UNSILENCE_FAIL),
+            (ctx, text_channel, text_role, False, silence.MSG_UNSILENCE_MANUAL),
+            (ctx, voice_channel, voice_role, True, silence.MSG_UNSILENCE_FAIL),
+            (ctx, voice_channel, voice_role, False, silence.MSG_UNSILENCE_MANUAL),
+        )
+
+        class PermClass:
+            """Class to Mock return permissions"""
+            def __init__(self, value: bool):
+                self.val = value
+
+            def __getattr__(self, item):
+                return self.val
+
+        for context, channel, role, permission, message in test_cases:
+            with mock.patch.object(channel, "overwrites_for", return_value=PermClass(permission)) as overwrites:
+                with self.subTest(channel=channel, message=message):
+                    await self.cog._unsilence_wrapper(channel, context)
+
+                    overwrites.assert_called_once_with(role)
+                    send_message.assert_called_once_with(message, ctx.channel, channel)
+
+                    send_message.reset_mock()
+
+    @mock.patch.object(silence.Silence, "_force_voice_sync")
+    @mock.patch.object(silence.Silence, "send_message")
+    async def test_correct_overwrites(self, send_message, _):
```

Comment:
As with the silence test, there seems to be redundancies here. I'm not sure why you rolled your own tests instead of re-using and modifying the existing tests for text channels.

Reply:
@HassanAbouelela This still needs to be addressed. I really think this test should be scrapped. To test overwrites, take inspiration from (or directly modify) `test_preserved_other_overwrites`, `test_restored_overwrites`, and `test_cache_miss_used_default_overwrites`.

Reply:
I believe I've duplicated all relevant tests now. db5b0751.